src/modules/auth/components/login/components/email-input.tsx

```diff
@@ -0,0 +1,31 @@
+import { useRef } from 'react';
+import InputBase from '@mui/material/InputBase';
+import Box from '@mui/material/Box';
+import { type LoginInputProps } from '#/modules/auth/types';
+import { isInputFieldEmpty } from '#/modules/auth/utils';
+import { inputStyles as styles } from '#/modules/auth/styles/input.sx';
+
+export function EmailInput({ focusedField, handleBlur, handleFocus, register }: LoginInputProps) {
+  const emailRef = useRef<HTMLInputElement>();
+  const { ref, ...rest } = register('email');
+
+  return (
+    <Box sx={styles(focusedField === 'email', false, focusedField === 'password').container}>
+      <Box sx={styles(focusedField === 'email', !isInputFieldEmpty(emailRef)).label}>Email</Box>
+
+      <InputBase
+        {...rest}
+        autoCorrect="off"
+        ref={(e: HTMLDivElement | null) => {
```

Comment:
I see a lot of repetition of this, lets create a separate function for that and maybe reuse it in the inputs too

Reply:
fixed